test/variant_get_test.cpp

```diff
@@ -356,6 +356,33 @@ inline void check_that_does_not_exist()
     check_that_does_not_exist_impl<var_req_t>();
 }
 
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+class MoveonlyType {
+public:
+    MoveonlyType() {}
+    ~MoveonlyType() {}
+
+    MoveonlyType(MoveonlyType&& other) {}
+    void operator=(MoveonlyType&& other) {}
+
+private:
+    MoveonlyType(const MoveonlyType&);
+    MoveonlyType(MoveonlyType&&);
+};
+
+inline void get_rvref_test()
+{
+  boost::variant<int, MoveonlyType> v;
+  v = MoveonlyType();
+  BOOST_CHECK(boost::get<MoveonlyType>(boost::move(v)));
+  BOOST_CHECK(boost::get<MoveonlyType>(v));
+
+  boost::relaxed_get<MoveonlyType&>(boost::variant<int, MoveonlyType>()) // Must compile
```

Comment:
Something is wrong here: this line must throw `bad_get()` because default constructed variant will hold an int.